include/clang/AST/Expr.h

```diff
@@ -4505,8 +4505,8 @@ class GenericSelectionExpr : public Expr {
 
 /// \brief Represents a Checked C bounds expression.
 class BoundsExpr : public Expr {
-private:
-  SourceLocation StartLoc, RParenLoc;
+  SourceLocation StartLoc, EndLoc;
+  friend class ASTStmtReader;
```

Comment:
Eek, friends.  Is this a standard clang pattern?  If we can't find a clean way to avoid doing this, I'd suggest that we at least clearly comment what it is that `ASTStmtReader` will be reaching in and grabbing/changing.